apps/smart-contracts/core/test/DepositHook.test.ts

```diff
@@ -342,4 +374,145 @@ describe('=> DepositHook', () => {
+
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(1)
+    })
+
+    it('sets score for two collections', async () => {
+      await depositHook
+        .connect(deployer)
+        .setCollectionScores([testERC721.address, secondERC721.address], [1, 2])
+
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(1)
+      expect(await depositHook.getCollectionScore(secondERC721.address)).to.eq(2)
+    })
+
+    it('sets score for same collection twice', async () => {
+      await depositHook.connect(deployer).setCollectionScores([testERC721.address], [1])
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(1)
+
+      await depositHook.connect(deployer).setCollectionScores([testERC721.address], [2])
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(2)
+    })
+
+    it('emits CollectionScoresChange', async () => {
+      const tx = await depositHook.connect(deployer).setCollectionScores([testERC721.address], [1])
+
+      await expect(tx)
+        .to.emit(depositHook, 'CollectionScoresChange')
+        .withArgs([testERC721.address], [1])
+    })
+  })
+
+  describe('# removeCollections', () => {
+    it('reverts if not role holder', async () => {
+      expect(
+        await depositHook.hasRole(await depositHook.REMOVE_COLLECTIONS_ROLE(), user.address)
+      ).to.eq(false)
+
+      await expect(depositHook.connect(user).removeCollections([])).revertedWith(
+        `AccessControl: account ${user.address.toLowerCase()} is missing role ${await depositHook.REMOVE_COLLECTIONS_ROLE()}`
+      )
+    })
+
+    it('removes a collection', async () => {
+      await depositHook.connect(deployer).setCollectionScores([testERC721.address], [1])
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(1)
+
+      await depositHook.connect(deployer).removeCollections([testERC721.address])
+
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(0)
+    })
+
+    it('removes two collections', async () => {
+      await depositHook
+        .connect(deployer)
+        .setCollectionScores([testERC721.address, secondERC721.address], [1, 1])
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(1)
+      expect(await depositHook.getCollectionScore(secondERC721.address)).to.eq(1)
+
+      await depositHook
+        .connect(deployer)
+        .removeCollections([testERC721.address, secondERC721.address])
+
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(0)
+      expect(await depositHook.getCollectionScore(secondERC721.address)).to.eq(0)
+    })
+
+    it('removes a collection which doesnt exist', async () => {
+      await depositHook.connect(deployer).setCollectionScores([testERC721.address], [1])
+      expect(await depositHook.getCollectionScore(testERC721.address)).to.eq(1)
```

Comment:
```suggestion
      expect(await depositHook.getCollectionScore(secondERC721.address)).to.eq(0)
```

No need to add the first collection, can just expect that it doesnt exist at the beginning, and subsequently verify that it still doesn't exist.

Since you didn't add an idempotence test here, i would actually just replace this with an idempotence test. So first add the collection, then remove it, then remove it again.